clang/lib/3C/ConstraintVariables.cpp

```diff
@@ -872,27 +878,24 @@ PointerVariableConstraint::mkString(Constraints &CS,
     case Atom::A_NTArr:
       if (emitArraySize(ConstArrs, TypeIdx, K))
         break;
-      AllArrays = false;
-      // This additional check is to prevent fall-through from the array.
-      if (K == Atom::A_NTArr) {
-        // If this is an NTArray.
-        getQualString(TypeIdx, Ss);
+      // If this is an NTArray.
+      getQualString(TypeIdx, Ss);
 
-        // We need to check and see if this level of variable
-        // is constrained by a bounds safe interface. If it is,
-        // then we shouldn't re-write it.
-        EmittedBase = false;
-        Ss << "_Nt_array_ptr<";
-        EndStrs.push_front(">");
-        break;
-      }
-      LLVM_FALLTHROUGH;
+      // We need to check and see if this level of variable
```

Comment:
I think this comment is completely obsolete. Back in 6c6f0d96999a944ff6c4174062f7ff50db7c22ae there was a `if (getItypePresent() == false) {` check here, but that got removed at some point without updating the comment.

Reply:
Yep, `getItypePresent` was renamed to `hasItype`, which was then removed in #506.  I removed the comments.